subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidateTaskProperties.java

```diff
@@ -307,26 +316,52 @@ public boolean getFailOnWarning() {
 
     /**
      * Returns the output file to store the report in.
+     *
+     * @since 4.5
      */
-    @Optional @OutputFile
-    public File getOutputFile() {
-        return outputFile == null ? null : getProject().file(outputFile);
+    @Optional
+    @OutputFile
+    public RegularFileProperty getOutputFile() {
+        return outputFile;
     }
 
     /**
      * Sets the output file to store the report in.
      *
      * @since 4.0
+     *
+     * @deprecated Use {@link #getOutputFile()} instead.
      */
+    @Deprecated
     public void setOutputFile(File outputFile) {
-        setOutputFile((Object) outputFile);
+        DeprecationLogger.nagUserOfReplacedMethod("ValidateTaskProperties.setOutputFile(File)", "getOutputFile()");
+        this.outputFile.set(outputFile);
     }
 
     /**
      * Sets the output file to store the report in.
+     *
+     * @deprecated Use {@link #getOutputFile()} instead.
      */
-    public void setOutputFile(Object outputFile) {
-        this.outputFile = outputFile;
+    @Deprecated
+    public void setOutputFile(@Nullable final Object outputFile) {
+        DeprecationLogger.nagUserOfReplacedMethod("ValidateTaskProperties.setOutputFile(Object)", "getOutputFile()");
+        this.outputFile.set(new AbstractProvider<RegularFile>() {
```

Comment:
I don't think you need/want to create a new subclass of Provider. If you want to create a Provider, use `Project.provider()`. You may also want to use `Project.layout.file`, which creates a `Provider<RegularFile>` from a `Provider<File>`.

Reply:
I am also not sure what we do in terms of DSL decoration for setters of `RegularFileProperty`s. Maybe we add already a setter like this?
I don't quite see the benefit in this setter anyway. Instead of allowing `Object` here as an argument I think we would be better off if the user provided `file(Object)` to this method. If they want to do something lazy, they can now use the `RegularFileProperty` anyway.

Reply:
This is solely to retain binary compatibility and to print the deprecation warning. I'm also fine breaking binary compatibility, because `@Incubating`.